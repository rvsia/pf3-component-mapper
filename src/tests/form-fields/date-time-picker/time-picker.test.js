import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import TimePicker from '../../../form-fields/date-time-picker/time-picker';

describe('<TimePicker />', () => {
  let initialProps;
  let initialDate = new Date('December 17, 2018 03:24:00');
  beforeEach(() => {
    initialProps = {
      onHourChange: jest.fn(),
      onMinuteChange: jest.fn(),
      selectedDay: initialDate,
    };
  });

  it('should render correcntly', () => {
    const wrapper = mount(<TimePicker { ...initialProps } />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should decrement minute by one on button click', () => {
    const onMinuteChange = jest.fn();
    const wrapper = mount(<TimePicker { ...initialProps } onMinuteChange={ onMinuteChange } />);
    wrapper.find('.chevron-button').last().simulate('click');
    expect(onMinuteChange).toHaveBeenCalledWith(initialDate.getMinutes() - 1);
  });

  it('should decrement minute to 59 if was 0 before on button click', () => {
    const onMinuteChange = jest.fn();
    const wrapper = mount(
      <TimePicker
        { ...initialProps }
        selectedDay={ new Date('December 17, 2018 03:00:00') }
        onMinuteChange={ onMinuteChange }
      />
    );
    wrapper.find('.chevron-button').last().simulate('click');
    expect(onMinuteChange).toHaveBeenCalledWith(59);
  });

  it('should increment minute by one on button click', () => {
    const onMinuteChange = jest.fn();
    const wrapper = mount(<TimePicker { ...initialProps } onMinuteChange={ onMinuteChange } />);
    wrapper.find('.chevron-button').at(1).simulate('click');
    expect(onMinuteChange).toHaveBeenCalledWith(initialDate.getMinutes() + 1);
  });

  it('should increment minute to 0 if was 59 on button click', () => {
    const onMinuteChange = jest.fn();
    const wrapper = mount(
      <TimePicker
        { ...initialProps }
        selectedDay={ new Date('December 17, 2018 03:59:00') }
        onMinuteChange={ onMinuteChange }
      />
    );
    wrapper.find('.chevron-button').at(1).simulate('click');
    expect(onMinuteChange).toHaveBeenCalledWith(0);
  });

  it('should decrement hours by one on button click', () => {
    const onHourChange = jest.fn();
    const wrapper = mount(<TimePicker { ...initialProps } onHourChange={ onHourChange } />);
    wrapper.find('.chevron-button').at(2).simulate('click');
    expect(onHourChange).toHaveBeenCalledWith(initialDate.getHours() - 1);
  });

  it('should decrement hours to 23 if was 0 before on button click', () => {
    const onHourChange = jest.fn();
    const wrapper = mount(
      <TimePicker
        { ...initialProps }
        selectedDay={ new Date('December 17, 2018 00:00:00') }
        onHourChange={ onHourChange }
      />
    );
    wrapper.find('.chevron-button').at(2).simulate('click');
    expect(onHourChange).toHaveBeenCalledWith(23);
  });

  it('should increment hours by one on button click', () => {
    const onHourChange = jest.fn();
    const wrapper = mount(<TimePicker { ...initialProps } onHourChange={ onHourChange } />);
    wrapper.find('.chevron-button').first().simulate('click');
    expect(onHourChange).toHaveBeenCalledWith(initialDate.getHours() + 1);
  });

  it('should increment hours to 0 if was 23 on button click', () => {
    const onHourChange = jest.fn();
    const wrapper = mount(
      <TimePicker
        { ...initialProps }
        selectedDay={ new Date('December 17, 2018 23:59:00') }
        onHourChange={ onHourChange }
      />
    );
    wrapper.find('.chevron-button').first(0).simulate('click');
    expect(onHourChange).toHaveBeenCalledWith(0);
  });

  it('should call minutes change callback on minute input change event', () => {
    const onMinuteChange = jest.fn();
    const wrapper = mount(<TimePicker { ...initialProps } onMinuteChange={ onMinuteChange } />);
    wrapper.find('input').last().simulate('change', { target: { value: 11 }});
    expect(onMinuteChange).toHaveBeenCalledWith(11);
  });

  it('should call hours change callback on hour input change event', () => {
    const onHourChange = jest.fn();
    const wrapper = mount(<TimePicker { ...initialProps } onHourChange={ onHourChange } />);
    wrapper.find('input').first().simulate('change', { target: { value: 5 }});
    expect(onHourChange).toHaveBeenCalledWith(5);
  });
});
