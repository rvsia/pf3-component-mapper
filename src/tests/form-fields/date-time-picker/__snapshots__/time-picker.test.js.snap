// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimePicker /> should render correcntly 1`] = `
<TimePicker
  onHourChange={[MockFunction]}
  onMinuteChange={[MockFunction]}
  selectedDay={2019-01-17T03:24:00.000Z}
>
  <table
    className="time-picker-layout-wrapper"
  >
    <tbody>
      <tr>
        <td
          className="chevron-button"
          onClick={[Function]}
        >
          <Icon
            name="chevron-up"
            type="fa"
          >
            <FontAwesome
              name="chevron-up"
            >
              <span
                aria-hidden={true}
                className="fa fa-chevron-up"
              />
            </FontAwesome>
          </Icon>
        </td>
        <td />
        <td
          className="chevron-button"
          onClick={[Function]}
        >
          <Icon
            name="chevron-up"
            type="fa"
          >
            <FontAwesome
              name="chevron-up"
            >
              <span
                aria-hidden={true}
                className="fa fa-chevron-up"
              />
            </FontAwesome>
          </Icon>
        </td>
      </tr>
      <tr>
        <td>
          <span>
            <input
              className="time-picker-hours-input"
              max={23}
              min={0}
              onChange={[Function]}
              type="number"
              value="04"
            />
          </span>
        </td>
        <td>
          :
        </td>
        <td>
          <span>
            <input
              className="time-picker-hours-input"
              max={59}
              min={0}
              onChange={[Function]}
              type="number"
              value="24"
            />
          </span>
        </td>
      </tr>
      <tr>
        <td
          className="chevron-button"
          onClick={[Function]}
        >
          <Icon
            name="chevron-down"
            type="fa"
          >
            <FontAwesome
              name="chevron-down"
            >
              <span
                aria-hidden={true}
                className="fa fa-chevron-down"
              />
            </FontAwesome>
          </Icon>
        </td>
        <td />
        <td
          className="chevron-button"
          onClick={[Function]}
        >
          <Icon
            name="chevron-down"
            type="fa"
          >
            <FontAwesome
              name="chevron-down"
            >
              <span
                aria-hidden={true}
                className="fa fa-chevron-down"
              />
            </FontAwesome>
          </Icon>
        </td>
      </tr>
    </tbody>
  </table>
</TimePicker>
`;
